main.js

```diff
@@ -3828,7 +3828,7 @@ function stopInstance(id, force, callback) {
                 } else if (!procs[id].startedAsCompactGroup) {
                     states.setState(id + '.sigKill', {val: -1, ack: false, from: hostObjectPrefix}, err => { // send kill signal
                         logger.info(hostLogPrefix + ' stopInstance ' + instance._id + ' send kill signal');
-                        if (!err) {
+                        if (!err && procs[id]) {
```

Comment:
Really there? in fact the callback should be called if setstate is ok,also if adapter got reomoved in the meantime... Having it that way the callback would not be called or?!

Reply:
it will then be called after the stop timeout https://github.com/ioBroker/ioBroker.js-controller/pull/896/files#diff-7a9076d6d94e62c13d641aa71f19ae8eL3856